azure-pipelines.yml

```diff
@@ -0,0 +1,67 @@
+# ASP.NET
+# Build and test ASP.NET projects.
+# Add steps that publish symbols, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/apps/aspnet/build-aspnet-4
+#trigger on main pull
+trigger:
+  branches:
+    include:
+    - main
+
+stages:
+- stage: build
+  jobs:
+  - job: build
+    pool:
+      vmImage: ubuntu-latest
+    variables:
+      sdkVersion: 5.x
+      buildConfiguration: 'Release'
+      directory: 'Fakebook.Posts/Fakebook.Posts.RestApi'
+
+    steps:
+    - task: UseDotNet@2
+      displayName: dotnet sdk 2.x
+      inputs:
+        packageType: 'sdk'
+        version: '2.x'
+        workingDirectory: '$(directory)'
+
+    - task: SonarCloudPrepare@1
+      inputs:
+        SonarCloud: 'SonarCloud Token'
+        organization: '2011-fakebook-project3'
+        scannerMode: 'MSBuild'
+        projectKey: '2011-fakebook-project3_posts'
+        extraProperties: |
+          sonar.exclusions=**/lib/**
+          sonar.cs.opencover.reportsPaths = $(Agent.TempDirectory)/**/coverage.opencover.xml
+          sonar.cs.vstest.reportsPaths=$(Agent.TempDirectory)/*.trx
+
+    - task: UseDotNet@2
+      displayName: 'Use .NET 5 SDK'
+      inputs:
+        packageType: 'sdk'
+        version: '5.x'
+        installationPath: $(Agent.ToolsDirectory)/dotnet
+        workingDirectory: 'Fakebook.Posts.RestApi/'
+    - task: Docker@2
+      displayName: docker build image
+      inputs:
+        repository: 'posts'
+        command: 'build'
+        Dockerfile: '**/Dockerfile'
+    - task: DotNetCoreCLI@2
+      displayName: dotnet test
+      inputs:
+        command: 'test'
+        arguments: '--configuration $(BuildConfiguration) --collect:"XPlat Code Coverage" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover'
```

Comment:
`BuildConfiguration` is not `buildConfiguration`, i've corrected this error in like 4 other code reviews, pay more attention.
really we could just hardcode Release everywhere and drop the variable but it's fine